pwnlib/util/proc.py

```diff
@@ -340,3 +340,593 @@ def wait_for_debugger(pid):
+            >>> m.flags.private
+            True
+            >>> m.private
+            True
+
+        Returns:
+            bool
+        """
+        return self.flags.private
+
+    @property
+    def shared(self):
+        """Shorcut for flags.shared
+
+        Examples:
+            >>> m = MemoryMap.from_str("55db09b78000-55db09b81000 rw-p 00114000 fe:01 9832010                    /usr/bins/bash")
+            >>> m.flags.shared
+            False
+            >>> m.shared
+            False
+
+        Returns:
+            bool
+        """
+        return self.flags.shared
+
+    def is_in_range(self, address):
+        """Indicates if the specified addr is in the range of the map
+
+        Arguments:
+            address (int): Memory address
+
+        Returns:
+            bool: True if addr is in the map memory range
+
+        Example:
+            >>> m = MemoryMap.from_str("55db09b78000-55db09b81000 rw-p 00114000 fe:01 9832010                    /usr/bins/bash")
+            >>> m.is_in_range(0x55db09b78000)
+            True
+            >>> m.is_in_range(0x55db09b81000)
+            False
+            >>> m.is_in_range(0x55db09b78200)
+            True
+        """
+        return self.start_address <= address < self.end_address
+
+    @classmethod
+    def from_str(cls, map_str):
+        """Retrieves a memory map from a string describing it, such as a line
+        of the file /proc/<pid>/maps.
+
+        Args:
+            map_str: String which describes a map
+
+        Returns:
+            MemoryMap
+
+        Example:
+            >>> m = MemoryMap.from_str("55db09b78000-55db09b81000 rw-p 00114000 fe:01 9832010                    /usr/bins/bash")
+            >>> str(m)
+            '55db09b78000-55db09b81000 rw-p 00114000 fe:01 9832010\\t\\t/usr/bins/bash'
+        """
+        parts = map_str.split()
+
+        start_address, end_address = parts[0].split("-")
+        start_address = int(start_address, 16)
+        end_address = int(end_address, 16)
+
+        flags = MemoryMapFlags.from_str(parts[1])
+
+        offset = int(parts[2], 16)
+
+        device_major, device_minor = parts[3].split(":")
+        device_major = int(device_major, 16)
+        device_minor = int(device_minor, 16)
+
+        inode = int(parts[4])
+
+        try:
+            path = parts[5]
+        except IndexError:
+            path = ""
+
+        return cls(start_address, end_address, flags, offset, device_major, device_minor, inode, path)
+
+    def __str__(self):
+        map_str = ""
+        map_str += "%x-%x" % (self.start_address, self.end_address)
+        map_str += " %s" % self.flags
+        map_str += " %08x" % self.offset
+        map_str += " %02x:%02x" % (self.device_major, self.device_minor)
+        map_str += " %d" % self.inode
+
+        if self.path:
+            map_str += "\t\t%s" % self.path
+
+        return map_str
+
+
+class MemoryMaps:
```

Comment:
Use `corefile.maps` or `corefile.mappings`, this is unnecessary

Reply:
Done in 0ca2072